system_setup/server/controllers/configure.py

```diff
@@ -69,13 +66,88 @@ async def configure(self, *, context):
 
             self.app.update_state(ApplicationState.POST_RUNNING)
 
+            dryrun = self.app.opts.dry_run
+            variant = self.app.variant
+            config = WSLConfigHandler(dryrun)
+            if variant == "wsl_setup":
+                wsl_identity = self.model.identity.user
+                if dryrun:
+                    log.debug("mimicking creating user %s",
+                              wsl_identity.username)
+                else:
+                    run_command(["/usr/sbin/useradd", "-m", "-s", "/bin/bash",
+                                 "-p", wsl_identity.password,
+                                 wsl_identity.username])
+                    run_command(["/usr/sbin/usermod", "-a",
+                                 "-c", wsl_identity.realname,
+                                 "-G", self.get_userandgroups(),
+                                 wsl_identity.username])
+            else:
+                wslconf_ad = self.model.wslconfadvanced.wslconfadvanced
+                config.update("WSL.automount.enabled",
```

Comment:
See my other comment, way too many file updates here, which can be fixed with the requested refactoring (with only one config.update call, passing both models)